### PR TITLE
New recommendation frontend

### PR DESCRIPTION
**Pull request by amesp73**

### CODE REVIEW

**Comment by amesp73** [comment]
making a PR just to make some comments on the code

File: react_frontend/src/Components/MovieCard.js [hunk 1]
```diff
@@ -2,34 +2,77 @@ import React, {useEffect, useState} from 'react'
...
+
+    const handleDeleteMovie = async (movie, watchlist) => {
         try {
 
             //Not sure why movie is a weird structure, but the correct way to access id is movie.movie.id
             // Make a request to delete a movie from a watchlist 
             const response = await axios.delete(`http://localhost:8080/watchlist/${watchlist.id}/${movie.movie.id}`)
             handleWatchlistUpdate();
 
+            fetchMovies();
```
Comment (amesp73) [inline react_frontend/src/Components/MovieCard.js:60]: I don't see why we need to fetchMovies() on delete? The handleWatchLIstUpdate() does a fetchWatchLists() in profile which setWatchlists(watchListData). That should make profile component rerender, which is parent of watchlist and moviecard so they all rerender with the latest watchlist info.

File: react_frontend/src/pages/Recommend.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+function Recommend({ user , movieRec }) {
+
+    //TODO: move this to Recommendations page
+    const handleRecommendMovie = async (movieRec) => {
+        try {
+
+            //Not sure why movie is a weird structure, but the correct way to access id is movie.movie.id
+            // Make a request to recommend a movie based on a movie id 
+            const response = await axios.get(`http://localhost:8080/recommendation/${movieRec.movie.id}`)
```
Comment (amesp73) [inline react_frontend/src/pages/Recommend.js:13]: If movieRec is the starting object you made in app, movieRec.movieObj makes sense I think. But when it changes to a number then number (like literally 100) doesn't have a movie key so movieRec.movie.id is going to error I think?

I think you basically start with movieRec being an object (see your useState in app.js), but it has a movieRec.movieObj, not movieRec.movie.id. When you click the recommend button you do a movieRec = movie.id, so then it turns into a number. I think this won't work either of those. Maybe just make it a number always?

File: react_frontend/src/pages/Recommend.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+        } catch (error) {
+            console.error('Error fetching recommendation:', error);
+        }
+    };
+
+
+    return (
+        <div id="parent">
+            <MovieCard movie={movieRec} />
```
Comment (amesp73) [inline react_frontend/src/pages/Recommend.js:24]: I don't think movieRec is a movie object. I think you need to use the response.data above to get the movie object. Not sure I haven't looked at recommend api but that's how the movie one works.

File: react_frontend/src/pages/Profile.js [hunk 4]
```diff
@@ -27,6 +30,9 @@ function Profile({ user }) {
     const handleWatchlistUpdate = () => {
         fetchWatchlists();
     };
+    let [movieRec, setMovieRec] = useState({
```
Comment (amesp73) [inline react_frontend/src/pages/Profile.js:33]: I think you actually did do the lift state thing, but you just change the movieRec variable from object to int and use it wrong in recommend page? I comment more below (making this comment after reading the other code lol) about what I mean by that 

I just did something kind of like recommend for the movie info page using location and state to go from search page to move info page (and telling movie info page what movie to use). I think what you have will work if you make movieRec make sense and rerender right, but location and state might be less code: https://github.com/corypride/teamStruggle/commit/d026f6ae9ffdee0f7d5b02222a1784880fa00555

**Comment by gbreternitz** [comment]
Closing as I decided not to have recommendations implemented in a separate page

**Pull request closed without merging**